pkg/processor/runtime/rpc/connection/connectionallocator.go

```diff
@@ -54,96 +55,221 @@ func NewConnectionAllocator(abstractConnectionManager *AbstractConnectionManager
-func (ca *ConnectionAllocator) Start() error {
-	ca.Logger.DebugWith("Starting connection allocator")
-	eventConnections := make([]*Connection, 0)
-	if ca.MinConnectionsNum != 0 {
-		for i := 0; i < ca.MinConnectionsNum; i++ {
-			var conn net.Conn
-			var err error
-
-			// Use retryable dial for the first connection
-			if i == 0 {
-				conn, err = retryableDial(ca.serverAddress, 30, 1*time.Second, 1*time.Minute)
-			} else {
-				conn, err = net.Dial("tcp", ca.serverAddress)
-			}
-
-			if err != nil {
-				return errors.Wrap(err, "Failed to establish connection")
-			}
-			eventConnections = append(eventConnections, NewConnection(ca.Logger, conn, ca))
-		}
-	}
-	// start event processing
-	for _, eventConnection := range eventConnections {
-		eventConnection.SetEncoder(ca.Configuration.GetEventEncoderFunc(eventConnection.Conn))
-		go eventConnection.AbstractEventConnection.RunHandler()
-	}
-
+func (ca *ConnectionAllocator) Start(pid int) error {
+	ca.pid = pid
+	ca.Logger.DebugWith("Starting connection allocator",
+		"wrapperPID", ca.pid)
+	// starts control message socket
 	if err := ca.startControlMessageSocket(); err != nil {
 		return errors.Wrap(err, "Failed to start control message socket")
 	}
-
-	// wait for start if required to
-	if ca.Configuration.WaitForStart {
-		ca.Logger.Debug("Waiting for start")
-		for _, eventConnection := range eventConnections {
-			eventConnection.WaitForStart()
-		}
-	}
-	eventProcessors := make([]eventprocessor.EventProcessor, len(eventConnections))
-	for i, eventConnection := range eventConnections {
-		eventProcessors[i] = eventConnection
-	}
-	if err := ca.allocator.SetObjects(eventProcessors); err != nil {
-		return errors.Wrap(err, "Failed to set objects in allocator")
+	if err := ca.startEventConnections(); err != nil {
+		return errors.Wrap(err, "Failed to start event connections")
 	}
 
-	ca.Logger.Debug("Connection allocator started")
+	ca.SetStatus(status.Ready)
 	return nil
-
 }
 
 func (ca *ConnectionAllocator) Stop() error {
-	var wg sync.WaitGroup
-
-	for _, eventConnection := range ca.allocator.GetObjects() {
-		connection := eventConnection.(Connection)
-		wg.Add(1)
+	ca.SetStatus(status.Stopping)
+	ca.Logger.DebugWith("Stopping connection allocator",
+		"wrapperPID", ca.pid)
 
-		go func() {
-			defer wg.Done()
-			if err := connection.Stop(); err != nil {
-				ca.Logger.WarnWith("Failed to close connection", "error", err)
-			}
-		}()
-	}
-
-	wg.Wait()
+	err := ca.stopAllocator()
 	ca.stopControlMessageSocket()
-	return nil
+	ca.Logger.DebugWith("Stopped connection allocator",
+		"wrapperPID", ca.pid)
+
+	ca.SetStatus(status.Stopped)
+	return err
 }
 
 func (ca *ConnectionAllocator) Allocate(duration time.Duration) (eventprocessor.EventProcessor, error) {
-	// TODO: support multiple connections
 	return ca.allocator.Allocate(duration)
 }
 
+func (ca *ConnectionAllocator) Release(connection eventprocessor.EventProcessor) {
+	// if the processor requires restart, recreate the connection
```

Comment:
This comment should be in the `status.RestartRequired` case, not here

Reply:
@TomerShor no, processor just should be renamed to connection

Reply:
but you comment about recreating on "restartRequired", then check `Stopping / Stopped`, so the comment is not in the right place.
And you have [this comment](https://github.com/nuclio/nuclio/pull/3553/files#diff-80aae3196e04eca4ba23924804e14be6f2f3f80a53f118c0787920ba52a2d6e4R114) too so I think the first one can just be removed